### ISSUE TITLE
Question about the returned "predicted_matrix"

### ISSUE THREAD

**Issue by Brandon-YuHu** [ViCCo-Group/frrsa#44]
Hello,

I would like to use the predicted matrix returned from your function, but there are some value returned as 9999 which means these pair are not predicted. But if k-fold cross-validation is applied, then each pair of conditions should be used as testing set during cross-validation. How would it be possible that some pairs have not been predicted? I am really confused and could you explain about this?

**Comment by PhilippKaniuth** [comment]
When conducting feature-reweighted RSA, what one basically does is trying to explain the overall dissimilarity of condition pairs from the target system with feature-specific (and reweighted) dissimilarities of those same pairs from the predicting system. To do so, one needs to estimate weights for each feature, i.e. betas in a regularized multiple regression framework. If one were to fit the regularized regression model to the whole dataset, that is, using all condition pairs available, and then judge how well this statistical model performs on the whole dataset one would overfit. Hence, one needs to split the data by for example conducting a k-fold cross-validation. This old news is just set the scene. Because the crucial part now is *how* one splits the dataset.

One could either pre-compute all possible condition pairs and split the *pairs* in training and test sets. That way, one would utilize the whole dissimilarity matrix and would eventually get predictions for all pairs. However, the off-diagonal cells of the lower (or upper) triangular part of a dissimilarity matrix are not independent! Imagine the dissimilarities of the pairs (A, B) and (A, C) were in the training set, then that would constrain the possible values that the dissimilarity of (B, C) in the test set could take (given the triangle inequality). This would lead to leakage: information about the test set would be present in the training set. This in turn would lead to the statistical model performing unfairly well on the test set.

Therefore, one needs to split the data based on *conditions* not pairs to avoid leakage. However, now, if conditions A, B, C, D are in the training set, and E, F, G are in the test set, then e.g. the pair A,E would never be used for either fitting or testing the model. If one repeats such a k-fold cross-validation a few times it could be that a few pairs never receive a predicted similarity. 

Hence, there is a trade-off between leakage and possibly not predicting  (dis-)similarity for some pairs. We opted for no leakage.
